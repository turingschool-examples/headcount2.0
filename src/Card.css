
.card {
  width: 150px;
  text-align: center;
  margin: 2%;
  background: white;
  opacity: .86;
  border-radius: 5px;
  padding: 10px;
  box-shadow: 3px 3px 3px #7c7c7c;
}

h1 {
  font-size: 20px;
}

.data {
  font-size: 15px;
  text-align: left;
  font-weight: bold;
}

.less-than {
  color: #ea6512;
}

.greater-than {
  color: #4f4f4f;
}
