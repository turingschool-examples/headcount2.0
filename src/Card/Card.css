.card {
    box-shadow: 5px 5px 8px;
    width: 185px;
    height: 233px;
    border: 1px #000000 solid;
    margin: 7px;
}

ul {
  list-style-type: none;
}

li {
  font-size: 12px;
}

h3 {
  text-align: center;
  font-size: 15px;
  padding: 0 7px;
}