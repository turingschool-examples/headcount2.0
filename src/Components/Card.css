.card {
  background-color: #375053;
  border-radius: 15px;
  box-shadow: 0px 0px 19px 1px rgba(0, 0, 0, .7);
  flex: 1 18em;
  list-style-type: none;
  margin: 0 2em 2em 2em;
  padding: 0 1em 1em 1em;
}

.selected {
  background-color: #5e7377;
  border: solid 4px #f7b488;
}

h2 {
  font-size: 2em;
  margin: .5em;
  text-align: center;
}

li {
  font-size: 1.2em;
  padding: 0 0 0 6em;
}

.stat {
  padding: 0 0 .5em 0;
}

.less-than {
  color: #f7b488;
}