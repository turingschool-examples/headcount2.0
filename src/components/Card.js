import React from 'react';
import PropTypes from 'prop-types';


const Card = ({ location, data, handleCompare, compare, active }) => {
  const classname = active === true ? 'card active' : 'card'
  return (
    <div className={classname} key={location} onClick={() => handleCompare(location)}>
      <h1 className='cardTitle'>
        {location}
      </h1>
      {Object.keys(data).map((key, index) => {


        return (
          <div className='cardData' key={index}>
            <li className={
              data[key] <= 0.5 ? 'negative' : 'positive'
            }> {key}: {data[key]} </li>
          </div>
        )
      })}
    </div>
  );
}

Card.propTypes = {
  location: PropTypes.string,
  data: PropTypes.object,
  handleCompare: PropTypes.func,
}

export default Card;
