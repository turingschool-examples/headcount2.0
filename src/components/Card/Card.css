/*CARD COMPONENT RELATED CSS GOES HERE*/
.Card {
  height: 350px;
  width: 250px;
  margin: 5px 0;
  padding: 0.5rem;
  padding-bottom: 1.25rem;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 4rem 1fr;
  justify-items: center;
  color: #006989;
  letter-spacing: 0.5px;
  border-radius: 5%;
  background: rgb(255, 255, 255);
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  box-sizing: border-box;
}

.Card h2 {
  text-align: center;
  grid-row: 1 / 2;
  font-family: "Roboto Condensed", sans-serif;
  font-weight: 700;
  font-size: 20px;
  margin: 0;
  padding-top: 1.25rem;
}

ul {
  grid-row: 2 / 3;
  list-style-type: none;
  font-family: "Roboto", sans-serif;
  font-weight: 400;
  text-size: 1.5rem;
  padding-left: 0;
  align-self: center;
}

.low {
  color: #EF713B;
}

.selected {
  border: 5px solid #EF713B;
}