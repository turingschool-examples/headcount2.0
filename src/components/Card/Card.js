import React from 'react';
import PropTypes from 'prop-types';

const Card = props => {
  const { location, stats } = props;

  const districtStats = Object.keys(stats).map(stat => <li key={stat}>{stats[stat]}</li>);

  return (
    <div>
      <h2>{location}</h2>
      <ul>
        {districtStats}
      </ul>
    </div>
  );
};

Card.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.object.isRequired
};

export default Card;
