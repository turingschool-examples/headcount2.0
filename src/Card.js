import React from 'react';
import './Card.css';

const Card = ({schoolNames, helper}) => {
  const schoolName = helper.data[schoolNames].location;
  const annualData = helper.data[schoolNames].annualData;
  console.log(annualData);
  return (
    <div className="card">
      <h1>{schoolName}</h1>
      {Object.keys(annualData).map((year, index) => {
        if (annualData[year] <= 0.5) {
          return (
            <ul>
              <li><p className="less-than data">{year}: {annualData[year]}</p></li>
            </ul>
          )
        } else {
          return (
            <ul>
              <li><p className="greater-than data">{year}: {annualData[year]}</p></li>
            </ul>
          )
        }
      })}
    </div>
  )
}

export default Card;
