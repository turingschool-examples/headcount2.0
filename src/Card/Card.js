import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Card.css';

class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      clicked: this.props.clickedState || false
    };
  }

  handleCardClick = () => {
    
    if (this.props.comparedCards.length === 2  && 
      !this.props.comparedCards.find(district => {
        return district.district.location === this.props.district.location;
      })) {
      return;
    }
    this.setState({ clicked: !this.state.clicked });
    this.props.updateCompareState(this.state.clicked, this.props);
  }

  render() {
    let districtKeys = Object.keys(this.props.district.stats);
    const listItems = districtKeys.map(( stat, index ) => {
      return (
        <li 
          key={ index }
          className={ this.props.district.stats[stat] > 0.5 ? 'greater-than' : 'less-than'}
        > { `${stat}: ${this.props.district.stats[stat]}` } </li>
      );
    });
  
    return (
      <div 
        className={this.state.clicked ? 'card clicked' : 'card'} 
        onClick={ this.handleCardClick }
      > 
        <h3>
          { this.props.district.location }
        </h3>
        <ul>
          { listItems }
        </ul>
      </div>
    );
  }
}

Card.propTypes = {
  district: PropTypes.string.isRequired,
  comparedCards: PropTypes.arrayOf(PropTypes.object),
  updateCompareState: PropTypes.func.isRequired,
  clickedState: PropTypes.bool
};



export default Card;