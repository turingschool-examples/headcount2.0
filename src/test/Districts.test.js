import React from 'react';
import Districts from '../components/Districts';
import Card from '../components/Card';
import { shallow } from 'enzyme';

describe('Districts', () => {
  let wrapper;
  const mockStats ={
    'COLORADO' : {
      location: 'COLORADO',
      stats: {},
      selected: false
    },
    'ARAPAHOE' : {
      location: 'ARAPAHOE',
      stats: {},
      selected: false
    }
  };

  beforeEach(() => {
    wrapper = shallow(<Districts stats={mockStats} setSelectedCard={jest.fn()} />);
  });
  
  it('should match the snapshot', () => {    
    expect(wrapper).toMatchSnapshot();
  });

  it('should render all the districts', () => {
    expect(wrapper.find(Card).length).toEqual(2);
  });  
});
