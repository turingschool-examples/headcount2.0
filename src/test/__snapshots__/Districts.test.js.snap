// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Districts should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Districts
    setSelectedCard={[Function]}
    stats={
        Object {
            "ARAPAHOE": Object {
              "location": "ARAPAHOE",
              "selected": false,
              "stats": Object {},
            },
            "COLORADO": Object {
              "location": "COLORADO",
              "selected": false,
              "stats": Object {},
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          location="COLORADO"
          selected={false}
          setSelectedCard={[Function]}
          stats={Object {}}
/>,
        <Card
          location="ARAPAHOE"
          selected={false}
          setSelectedCard={[Function]}
          stats={Object {}}
/>,
      ],
      "className": "districtCards",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0district",
        "nodeType": "function",
        "props": Object {
          "location": "COLORADO",
          "selected": false,
          "setSelectedCard": [Function],
          "stats": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1district",
        "nodeType": "function",
        "props": Object {
          "location": "ARAPAHOE",
          "selected": false,
          "setSelectedCard": [Function],
          "stats": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            location="COLORADO"
            selected={false}
            setSelectedCard={[Function]}
            stats={Object {}}
/>,
          <Card
            location="ARAPAHOE"
            selected={false}
            setSelectedCard={[Function]}
            stats={Object {}}
/>,
        ],
        "className": "districtCards",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0district",
          "nodeType": "function",
          "props": Object {
            "location": "COLORADO",
            "selected": false,
            "setSelectedCard": [Function],
            "stats": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1district",
          "nodeType": "function",
          "props": Object {
            "location": "ARAPAHOE",
            "selected": false,
            "setSelectedCard": [Function],
            "stats": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
